File: 0876-hand-of-straights/0876-hand-of-straights.cpp
class Solution {
public:
    bool isNStraightHand(vector<int>& hand, int groupSize) {
        int n = hand.size();
        if(n%groupSize) return false;
        map<int, int> mp;
        for(int i=0; i<n; i++) mp[hand[i]]++;
        while(!mp.empty()){
            int currCard = mp.begin()->first;
            for(int i=0; i<groupSize; i++){
                if(mp[currCard+i]==0) return false;
                mp[currCard+i]--;
                if(mp[currCard+i]<1) mp.erase(currCard+i);
            }
        }
        return true;
    }
};

File: 0876-hand-of-straights/0876-hand-of-straights.java
class Solution {
    public boolean isNStraightHand(int[] hand, int groupSize) {
        int n = hand.length;
        if(n%groupSize != 0) return false;
        Map<Integer, Integer> mp = new TreeMap<>();
        for(int i=0; i<n; i++) mp.put(hand[i], mp.getOrDefault(hand[i], 0)+1);
        while(mp.size() > 0){
            int currCard = mp.entrySet().iterator().next().getKey();
            for(int i=0; i<groupSize; i++){
                if(!mp.containsKey(currCard+i)) return false;
                mp.put(currCard+i, mp.get(currCard+i)-1);
                if(mp.get(currCard+i)==0) mp.remove(currCard+i);
            }
        }
        return true;
    }
}